name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DZBankGruppePDFExtractor.java

```diff
@@ -804,26 +806,40 @@ private void addTaxReturnBlock(Map<String, String> context, DocumentType type)
                 .assign((t, v) -> processFeeEntries(t, v, type))
 
                 // Anlage 2.125,00 2,00 113,45 18,730
-                .section("amount", "feeFx").optional()
-                .match("^Anlage (?<amount>[\\.,\\d]+) (?<feeFx>[\\.,\\d]+) [\\.,\\d]+ [\\.,\\d]+")
+                .section("fxAmount", "fxFee").optional()
+                .match("^Anlage (?<fxAmount>[\\.,\\d]+) (?<fxFee>[\\.,\\d]+) [\\.,\\d]+ [\\.,\\d]+$")
                 .assign((t, v) -> {
-                    // Fee in percent
-                    double amount = Double.parseDouble(v.get("amount").replace(".", "").replace(",", "."));
-                    double feeFx = Double.parseDouble(v.get("feeFx").replace(",", "."));
-                    String fee =  Double.toString((amount / (1 + feeFx / 100)) * (feeFx / 100)).replace(".", ",");
+                    BigDecimal fxFeeDivisor1 = asExchangeRate(v.get("fxFee")).divide(new BigDecimal(100));
+                    BigDecimal fxFeeDivisor2 = fxFeeDivisor1.add(BigDecimal.ONE);
+
+                    // fxFee = (fxAmount / (1 + fxFee / 100)) * (fxFee / 100)
+                    BigDecimal fxFee = asExchangeRate(v.get("fxAmount")).divide(fxFeeDivisor2, 10,
+                                    RoundingMode.HALF_DOWN);
+                    fxFee = fxFee.multiply(fxFeeDivisor1).setScale(2, RoundingMode.HALF_UP);
+
+                    String fee = (String) NumberFormat.getNumberInstance(Locale.GERMANY)
+                                    .format(fxFee);
+
```

Comment:
Ein paar Überlegungen hierzu:
* Erstens: so richtig lesbarer wird der Code hier nicht (mit fxFeeDivisor1 und 2)
* Zweitens: ich finde es unschön - wenn wir schon die Zahl haben - die in einen String umzuwandeln nur um die im nächsten Schritt wieder in eine Zahl umzuwandeln. Ich glaube hier kann man direkt die Methoden im PDFExtractorUtils aufrufen

Was hälst Du davon?

```java
                // Anlage 2.125,00 2,00 113,45 18,730
                .section("fxAmount", "percentage").optional()
                .match("^Anlage (?<fxAmount>[\\.,\\d]+) (?<percentage>[\\.,\\d]+) [\\.,\\d]+ [\\.,\\d]+$")
                .assign((t, v) -> {
                    BigDecimal percentage = asBigDecimal(v.get("percentage"));
                    if (percentage.compareTo(BigDecimal.ZERO) != 0)
                    {
                        BigDecimal amount = asBigDecimal(v.get("fxAmount"));
                        BigDecimal fee = amount.divide(percentage.add(BigDecimal.valueOf(100)), Values.MC)
                                        .multiply(percentage, Values.MC);

                        Money feeMoney = Money.of(CurrencyUnit.EUR, fee.movePointRight(2)
                                        .setScale(0, Values.MC.getRoundingMode()).longValue());
                        checkAndSetFee(feeMoney, t, type);
                    }
                })
```